### ISSUE TITLE
train_hybrid.py: --activation leakyrelu (pytorch) regresses argmin_acc 12pp vs --activation relu (sklearn) on identical hybrid-heads data

### ISSUE THREAD

**Issue by lilith** [imazen/zenanalyze#68]
## TL;DR

The `--activation` flag is documented as "produces a `student.coefs_/intercepts_` surface so safety_check, diagnostics, and JSON serialization work the same way" — implying the choice is a speed/numerics knob, not a generalization knob. Empirically on the zenwebp v0.2 6-cell × 4-scalar hybrid-heads training, the pytorch `leakyrelu` path loses **12pp of argmin_acc** vs the sklearn `relu` path at identical architecture and identical training/val splits.

The leakyrelu speed advantage is real (~10×), but the documented framing undersells the generalization tradeoff. Worth a per-backend metric in the safety_report so codec teams notice this when it bites.

## Reproduction

zenwebp v0.2 sweep (2026-05-01): 6 cells × (1 + 4 scalar heads) = 30 outputs, 12,788 decision rows, 78-input MLP, hidden=128,128,128, seed=0xCAFE, holdout_frac=0.20 (image-level split).

Same `--codec-config zenwebp_picker_config`, same data, same seed, same hidden shape. Only `--activation` differs:

```
--activation leakyrelu (pytorch backend):
  Student argmin mean overhead: 2.76%
  Student argmin acc:           30.6%
  train→val gap:                +0.10pp

--activation relu (sklearn backend):
  Student argmin mean overhead: 2.68%
  Student argmin acc:           43.0%   ← +12.4pp better
  train→val gap:                -0.00pp ← actually slightly negative
  trained final loss=3.04, n_iter=500 (early-stopped)
```

Both teachers identical (HistGB-derived soft targets, same code path): mean overhead 2.62% / argmin_acc 40.1%. So the divergence is entirely in the student MLP.

## Why this matters

- **Codec teams using `--activation leakyrelu` for the speed win silently ship a worse picker.** Argmin acc gap of 12pp at this scale corresponds to ~+0.5-1.0pp mean byte overhead in production — meaningful for SLA-bound traffic.
- **The current `--help` text and `tools/README.md` recommend leakyrelu** for new bakes ("10–20× faster"). They're factually correct on speed but mislead on generalization. The recommendation should be conditional: leakyrelu for iteration / capacity sweeps; sklearn relu for production bakes (until the gap is understood and closed).

## Hypothesis 1: optimizer / init differences

`_train_torch_leakyrelu_student` (in `zentrain/tools/train_hybrid.py`) uses default PyTorch init (Kaiming-uniform on `nn.Linear` — appropriate for ReLU-family) and a plain `Adam(lr=2e-3)` with `MSELoss`. sklearn's `MLPRegressor` uses Glorot-uniform + Adam with `learning_rate_init=2e-3` and the same MSE — but its training loop has a built-in `ConvergenceWarning` early-stopping schedule with `n_iter_no_change=30` that may stop earlier on a noisier validation slice.

The pytorch path's hold-out internal val slice (`val_idx`) for early stopping is probably tighter than sklearn's, which converges on a different optimum.

## Hypothesis 2: weight regularization

sklearn's `MLPRegressor` defaults to `alpha=1e-4` L2 penalty on weights. The pytorch path has no L2 penalty in the optimizer setup (no `weight_decay` argument to `optim.Adam`). The pytorch student is unregularized; on this data scale it overfits less obviously than sklearn (gap is only +0.10pp on val), but the unregularized weight matrix doesn't generalize.

If hypothesis 2 is correct, adding `weight_decay=1e-4` to the pytorch Adam call should close most of the gap.

## Proposed work

1. **Confirm the gap** on a second codec config (zenjpeg v2.2). If sklearn relu also wins by 5-15pp argmin acc there, this is a generalizable issue.
2. **Add `weight_decay=1e-4`** to the pytorch Adam call (`_train_torch_leakyrelu_student`). Re-run zenwebp v0.2 6c4s. If the gap closes to ≤2pp, ship as the default and update docs.
3. **Add a backend-marker to `safety_report`** so the bake JSON records which trainer was used. Codec runtime can then differentiate v0.1.x bakes from v0.2.x bakes if behavior diverges.
4. **Update the `--activation` docstring + `--help`** to acknowledge the generalization tradeoff: "sklearn-relu is the recommended default for production bakes; leakyrelu (pytorch) is 10-20× faster but may regress argmin_acc — use for capacity sweeps + ablation passes, not final bakes."

## Linked

- imazen/zenanalyze#67 — per-head scalar normalization (different bug, same trainer; both bite hybrid-heads pickers).
